### PR TITLE
Add spray can, fix GPU instantiation for decals

### PR DESCRIPTION
**Pull request by Fueredoriku**
- Added a spray can with modifiable spray color!
- Made the existing bullet decals use GPU instantiation (yikes, atleast we should see a decent performance increase now)
- Fixed playerhands being in the wrong lightlayer for avoiding decals

Note: I'm not adding the spray cans to the maps in this PR, as I plan to add more props very soon and want to avoid conflicts
sprayCans

### CODE REVIEW

File: Assets/Scripts/Interactables/SprayCan.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    private void StartSpraying()
+    {
+        state = SprayCanState.Spraying;
+        sprayEffect.SendEvent(VisualEffectAsset.PlayEventID);
+        sprayEffect.SetBool("IsSpraying", true);
+        audioSource.loop = true;
+        sprayNoise.PlayExclusively(audioSource);
+    }
+
+    private void StopFlying()
```
Comment (toberge) [inline Assets/Scripts/Interactables/SprayCan.cs]: StopSpraying?

File: Assets/Scripts/Interactables/SprayCan.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        }
+        timeSinceDecalSpawn = 0;
+
+        if (Physics.Raycast(transform.position, transform.forward, out RaycastHit info, 1f))
+        {
+
+            // Avoid placing decals on players or their guns, as that leads to the heebie-jeebies
+            if (((1 << info.collider.gameObject.layer) & allHitboxesAndGunsAndPlayersMask) > 0)
+                return;
+            // Specifically avoid players. This is still required, even with the layer check above.
```
Comment (toberge) [inline Assets/Scripts/Interactables/SprayCan.cs:123]: Migt be useful to share code for the decal guard, but no biggie yet

**Review by toberge — APPROVED** [review]
LGTM

File: Assets/Scripts/Interactables/SprayCan.cs (hunk 2 above)
Comment (Fueredoriku) [reply Assets/Scripts/Interactables/SprayCan.cs:123]: Yeah, we might fix it *someday* 😅

File: Assets/Scripts/Interactables/SprayCan.cs (hunk 1 above)
Comment (Fueredoriku) [reply Assets/Scripts/Interactables/SprayCan.cs]: Fixed 👍

**Pull request merged**